lib/broccoli/ember-app.js

```diff
@@ -1073,19 +1073,20 @@ class EmberApp {
    * @return {BroccoliTree}
   */
   getStyles() {
-    let styles;
+    let styles = [];
     if (this.trees.styles) {
-      styles = new Funnel(this.trees.styles, {
-        srcDir: '/',
-        destDir: '/app/styles',
-        annotation: 'Funnel (styles)',
-      });
+      styles.push(this.trees.styles);
     }
     let addons = this.addonTreesFor('styles');
-
-    return mergeTrees(addons.concat(styles), {
+    styles = styles.concat(addons);
+    let mergedAddonStyles = mergeTrees(styles, {
       overwrite: true,
-      annotation: 'Styles',
+      annotation: 'Addon Styles',
+    });
+
+    return new Funnel(mergedAddonStyles, {
+      allowEmpty: true,
+      destDir: isExperimentEnabled('MODULE_UNIFICATION') ? 'src/ui/styles' : `app/styles`,
```

Comment:
In other places is checking for the existence of the `src` folder like:

`isExperimentEnabled('MODULE_UNIFICATION') && this.trees.src`